docs/guides/fetch-data/02-read-asset-data.md

```diff
@@ -0,0 +1,1056 @@
+const sampleUniversalReceiver = '0x50a02ef693ff6961a7f9178d1e53cc8bbe1dad68';
+
+// ABI for the Universal Receiver
+const lsp1MinimalInterface = [
+  {
+    inputs: [],
+    name: 'getAllRawValues',
+    outputs: [
+      {
+        internalType: 'bytes32[]',
+        name: '',
+        type: 'bytes32[]',
+      },
+    ],
+    stateMutability: 'view',
+    type: 'function',
+  },
+];
+
+/*
+ * Return array of blockchain addresses of
+ * assets, that were received by the
+ * Univeral Profile holder.
+ *
+ * @returns: array of asset addresses
+ */
+async function getAssetAddressses() {
+  const AddressStore = new web3.eth.Contract(
+    lsp1MinimalInterface,
+    sampleUniversalReceiver,
+  );
+
+  let rawValues = [];
+
+  try {
+    // Fetch all raw values
+    rawValues = await AddressStore.methods.getAllRawValues().call();
+  } catch (error) {
+    return console.log('Data from universal receiver could not be loaded');
+  }
+
+  let digitalAssets = [];
+
+  // Retrieve addresses
+  for (let i = 0; i < rawValues.length; i++) {
+    digitalAssets[i] = web3.utils.toChecksumAddress(rawValues[i].substr(26));
+  }
+  return digitalAssets;
+}
+
+// Debug
+getAssetAddressses().then((digitalAssets) => console.log(digitalAssets));
+```
+
+## Step 2 - Check ownership of assets
+
+After trimming out the asset addresses, we can check which assets are owned by the Universal Profile. We can do this by comparing the balances of the assets within the receiver contract. If the balance is greater than zero, the asset is still owned.
+
+```javascript title="read_assets.js"
+...
+// Our static Universal Profile address
+const sampleProfileAddress = '0x0C03fBa782b07bCf810DEb3b7f0595024A444F4e';
+
+// ABI for the asset
+const lsp8MinimalInterface = [
+  {
+    inputs: [
+      {
+        internalType: "address",
+        name: "tokenOwner",
+        type: "address",
+      },
+    ],
+    name: "balanceOf",
+    outputs: [
+      {
+        internalType: "uint256",
+        name: "",
+        type: "uint256",
+      },
+    ],
+    stateMutability: "view",
+    type: "function",
+  },
+];
+
+/*
+ * Return array of blockchain addresses
+ * of assets that are owned by the
+ * Univeral Profile.
+ *
+ * @returns: array of asset addresses
+ */
+async function getOwnedAddresses(owner) {
+  let digitalAssets = await getAssetAddressses();
+  let ownedAssets = [];
+
+  for (let i = 0; i < digitalAssets.length; i++) {
+    let isCurrentOwner;
+
+    // Create instance of the asset to check owner balance
+    const assetContract = new web3.eth.Contract(
+      lsp8MinimalInterface,
+      digitalAssets[i]
+    );
+
+    isCurrentOwner = await assetContract.methods.balanceOf(owner).call();
+    if (isCurrentOwner > 0) {
+      ownedAssets[ownedAssets.length] = digitalAssets[i];
+    }
+  }
+  return ownedAssets;
+}
+
+// Debug
+getOwnedAddresses(sampleProfileAddress).then((ownedAssets) => console.log(ownedAssets));
+```
+
+## Step 3 - Check the asset's interface
```

Comment:
I would put this step before (as Step 2) + rename it to:

"Check the type of asset"

Reply:
To get the balance of an asset, it does not matter which interface is behind in terms of LSP7/LSP8. I would leave this structure because you fetch the interface (step 3) right before it is needed (step 4) instead of doing a task that has nothing to do with the following step and might be confusing.